keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DockerArtifactSupplier.kt

```diff
@@ -69,46 +69,20 @@ class DockerArtifactSupplier(
 
     // Note: we currently don't have a way to derive account information from artifacts,
     // so, in the calls to clouddriver below, we look in all accounts.
-    val latestTag = runWithIoContext {
+    return runWithIoContext {
       cloudDriverService
         .findDockerTagsForImage("*", artifact.name, deliveryConfig.serviceAccount)
         .distinct()
-        .sortedWith(artifact.versioningStrategy.comparator)
+        // FIXME: this is making N calls to fill in data for each version so we can sort.
+        //  Ideally, we'd make a single call to return the list with details for each version.
+        .mapNotNull { getArtifactByVersion(artifact, it) }
+        .sortedWith(artifact.sortingStrategy.comparator)
         .firstOrNull()
     }
-
-    return if (latestTag != null) {
```

Comment:
why was this logic deleted? where are we creating the newly published artifact if its not the same?

Reply:
Because I changed the code immediately above to retrieve the details for all the versions, not just for the latest, so we can sort with the new `SortingStrategy`. The list now has the full objects, not just the version strings.